### PR TITLE
Add support for GCE automatic subnetwork configuration

### PR DESCRIPTION
**Pull request by ginkel**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

cloud/gce_net
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel fdf22b5a40) last updated 2016/07/07 14:42:07 (GMT +200)
  lib/ansible/modules/core: (gce-auto-subnetting b362f9877b) last updated 2016/07/07 20:32:58 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 0f01acd6c4) last updated 2016/07/07 14:42:18 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This PR adds support for auto subnetwork configuration to the `gce_net` module. For details, see https://cloud.google.com/compute/docs/subnetworks ("Auto subnet network").

Using this feature requires libcloud >= 1.0.0, but the implementation degrades gracefully if an older libcloud is present, but the feature is not used.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ginkel. To the current maintainers, @erjohnso please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/google/gce_net.py [hunk 1]
```diff
@@ -221,12 +230,22 @@ def main():
...
+            if not ipv4_range and mode == 'legacy':
+                module.fail_json(msg="Network '" + name + "' is not found. To create network in legacy mode, 'ipv4_range' parameter is required",
                     changed=False)
 
             try:
-                network = gce.ex_create_network(name, ipv4_range)
+                if mode != 'legacy':
+                    import inspect
+
+                    args = inspect.getargspec(gce.ex_create_network)
```
Comment (supertom) [inline cloud/google/gce_net.py:241]: Could you do the version check[ like this](https://github.com/tgbyte/ansible-modules-core/blob/5be9f4955136ed0679144c8e99e7cc129a3da7ef/cloud/google/gce.py#L598)?  The argument to that method is in v1.0 and is expected to remain.

File: cloud/google/gce_net.py [hunk 2]
```diff
@@ -55,6 +55,13 @@
     required: false
     default: null
     aliases: []
+  mode:
+    version_added: "2.2.0"
+    description:
+      - network mode (auto or legacy)
+    required: false
+    default: legacy
```
Comment (supertom) [inline cloud/google/gce_net.py:63]: default: "legacy"

Comment (supertom) [inline cloud/google/gce_net.py:63]: Could you add a 'choices' section here, too?

**Comment by supertom** [comment]
Thanks @ginkel for the PR!  Left you some comments.

**Comment by gregdek** [comment]
Thanks @ginkel for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@ginkel A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**